UI/window-basic-main-outputs.cpp

```diff
@@ -2011,22 +2011,23 @@ bool AdvancedOutput::ReplayBufferActive() const
 
 /* ------------------------------------------------------------------------ */
 
-bool BasicOutputHandler::SetupAutoRemux(const char *&ext)
+void BasicOutputHandler::SetupAutoRemux(const char *&ext)
 {
 	bool autoRemux = config_get_bool(main->Config(), "Video", "AutoRemux");
 	if (autoRemux && strcmp(ext, "mp4") == 0)
 		ext = "mkv";
-	return autoRemux;
 }
 
 std::string
 BasicOutputHandler::GetRecordingFilename(const char *path, const char *ext,
 					 bool noSpace, bool overwrite,
 					 const char *format, bool ffmpeg)
 {
-	bool remux = !ffmpeg && SetupAutoRemux(ext);
+	if (!ffmpeg)
+		SetupAutoRemux(ext);
+
 	string dst = GetOutputFilename(path, ext, noSpace, overwrite, format);
-	lastRecordingPath = remux ? dst : "";
+	lastRecordingPath = dst;
```

Comment:
This makes me concerned, because I'm almost totally positive there's a reason why this was the way it was. It would be unsafe to assume otherwise.

Reply:
Please check e5d8f345fc (both the commit message and the code) to examine why this was the way it was and whether it's safe to do this.

Reply:
The reason why I do this, is because the lastRecordingPath was set only when auto remuxing. Now it is set with every recording. Before https://github.com/obsproject/obs-studio/commit/e5d8f345fc30524d01942e1f5677343101ffb5be,  the lastRecordingPath was in the GetOutputFilename function, which the screenshots also used.

With https://github.com/obsproject/obs-studio/pull/3953/files#diff-4c0a52244f264feca9c3031c90edcbbbd1087a3040c15f9df4ca9700c5154fa3R5918, the auto remux function checks if auto remuxed is enabled instead of checking if the string is empty.